[doc] Changes entry for #7483 + doc improvements (#8202)
ocaml/dune: doc/howto/opam-file-generation.rst

If you have existing ``*.opam`` files, make a backup of them because the instructions in this section will overwrite them.

Now that you have declared package metadata in ``dune-project``, you can add
``(generate_opam_files)`` in ``dune-project``.

From now on, commands like ``dune build`` and ``dune runtest`` are going to regenerate the contents of opam files from the metadata in ``(package)`` stanzas.

Run ``dune build`` once and observe that the opam files have been created or
updated. Make sure to add these changes to your version control system.

You can control if ``.opam`` files will be generated at the root of
the project, or under an ``opam`` directory with the
:dune:ref:`opam_file_location` stanza in
``dune-project``. ``(opam_file_location inside_opam_directory)`` will
place the files inside the ``${project_root}/opam``, and
``(opam_file_location relative_to_project)`` will generate them at the

> root of the project, as it has been the case since the Opam file
> generation feature was introduced.

Reviewer: ```suggestion
root of the project, which is the default.
```
